x-pack/plugin/searchable-snapshots/src/test/java/org/elasticsearch/xpack/searchablesnapshots/AbstractSearchableSnapshotsRestTestCase.java

```diff
@@ -96,20 +115,71 @@ private void runSearchableSnapshotsTest(SearchableSnapshotsTestCaseBody testCase
         }
 
         final String indexName = randomAlphaOfLength(10).toLowerCase(Locale.ROOT);
-        final int numberOfShards = randomIntBetween(1, 5);
-
         logger.info("creating index [{}]", indexName);
         createIndex(
             indexName,
-            Settings.builder()
-                .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, numberOfShards)
-                .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
-                .build()
+            indexSettings != null
+                ? indexSettings
+                : Settings.builder()
+                    .put(IndexMetadata.SETTING_NUMBER_OF_SHARDS, randomIntBetween(1, 5))
+                    .put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, 0)
+                    .build(),
+            """
+                    "properties": {
+                        "field": {
+                            "type": "integer"
+                        },
+                        "text": {
+                            "type": "text",
+                            "fields": {
+                                "raw": {
+                                    "type": "keyword"
+                                }
+                            }
+                        }
+                    }
+                """
         );
         ensureGreen(indexName);
 
-        final int numDocs = randomIntBetween(1, 500);
         logger.info("indexing [{}] documents", numDocs);
+        final int indexingThreads = 2;
+        final CountDownLatch indexingLatch = new CountDownLatch(indexingThreads);
+        final AtomicLong remainingDocs = new AtomicLong(numDocs);
+        for (int i = 0; i < indexingThreads; i++) {
```

Comment:
I think we should remove the code that indexes `numDocs` below?

Reply:
Thanks for catching this, I pushed 0954893c009a93d528510ec92edd5b8ba6a5f737